connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoaderTest.java

```diff
@@ -61,4 +69,59 @@ public void testLoadingPluginClass() throws ClassNotFoundException {
             assertNotNull(classLoader.pluginClassLoader(pluginClassName));
         }
     }
+
+    @Test
+    public void testLoadingInvalidUberJar() throws Exception {
+        pluginPath.newFile("invalid.jar");
+
+        DelegatingClassLoader classLoader = new DelegatingClassLoader(
+            Collections.singletonList(pluginPath.getRoot().getAbsolutePath()));
+        classLoader.initLoaders();
+    }
+
+    @Test
+    public void testLoadingPluginDirContainsInvalidJarsOnly() throws Exception {
+        pluginPath.newFolder("my-plugin");
+        pluginPath.newFile("my-plugin/invalid.jar");
+
+        DelegatingClassLoader classLoader = new DelegatingClassLoader(
+            Collections.singletonList(pluginPath.getRoot().getAbsolutePath()));
+        classLoader.initLoaders();
+    }
+
+    @Test
+    public void testLoadingNoPlugins() throws Exception {
+        DelegatingClassLoader classLoader = new DelegatingClassLoader(
+            Collections.singletonList(pluginPath.getRoot().getAbsolutePath()));
+        classLoader.initLoaders();
+    }
+
+    @Test
+    public void testLoadingPluginDirEmpty() throws Exception {
+        pluginPath.newFolder("my-plugin");
+
+        DelegatingClassLoader classLoader = new DelegatingClassLoader(
+            Collections.singletonList(pluginPath.getRoot().getAbsolutePath()));
+        classLoader.initLoaders();
+    }
+
+    @Test
+    public void testLoadingMixOfValidAndInvalidPlugins() throws Exception {
+        TestPlugins.assertAvailable();
+
+        pluginPath.newFile("invalid.jar");
+        pluginPath.newFolder("my-plugin");
+        pluginPath.newFile("my-plugin/invalid.jar");
+        for (String fullPath : TestPlugins.pluginPath()) {
+            FileUtils.copyFileToDirectory(new File(fullPath), pluginPath.getRoot());
```

Comment:
Great tests. 
I'd suggest using nio (`Files.copy` and `Paths`), rather than bringing in the dependency in the `org.codehaus.plexus` util, that is available now, but is not listed explicitly in the project dependencies. For simple things like these it's almost always better to use standard java library stuff in tests.

Reply:
Good point. And, in fact, the import is disallowed. noice.
```
DelegatingClassLoaderTest.java:20:1: Disallowed import - org.codehaus.plexus.util.FileUtils. [ImportControl]
```